doc/intro.rst

```diff
@@ -399,3 +399,63 @@ Example
           - name: LoggingLevelDefault
           - name: LoggingTargetsDefault
           ...
+
+
+Validation
+----------
+
+The conformance to a specification can be checked against a Tango device.
+When the optional `-bidirectional` flag is added the differences that are on the device
+and not in the specification are also listed.
+
+.. code-block:: bash
+
+    $ tango-yaml validate -h
+
+      usage: tango_yaml validate [-h] (-url URL | -path PATH) [-bidirectional]
+                                tango_device_name
+
+      positional arguments:
+        tango_device_name  Tango device name in the format domain/family/member.
+                          TANGO_HOST env variable has to be set
+
+      optional arguments:
+        -h, --help         show this help message and exit
+        -url URL           The URL to a YAML specification file
+        -path PATH         The file path to a YAML specification file
+        -bidirectional     When bidirectional is included, any details on the device
+                          that is not in the spec is also listed.
+
+Example
+
+.. code-block:: bash
+
+    $ tango-yaml validate -path ./DishMaster.yaml  mid_d0001/elt/master
+
+      Command differs, [SetDSStandbyFPModeTask] specified but missing in device
+      Command [GetVersionInfo] differs:
+              doc_out:
```

Comment:
I guess the documentation fields (doc_in, doc_out, description) are very likely to differ in practice.  The data types and names are more important for interface validation.  It would be nice to have an option to ignore some keys. Probably better to add that in a future, once we are using it and it turns out to be a problem.

Reply:
Yeah, it's formatting can be weird. I split the "specification" and "device" in separate lines, but if the doc is large it can be difficult (but not impossible) to tell it apart.
Example below:

```
Command [SetStowMode] differs:
        doc_out:
                specification: == Long field ==
[0]: ack code (0=OK, 1=QUEUED, 2=FAILED)

== String field ==
[0]: Informative message or error/fault description
                device: Uninitialised
```